File: AspNetMvcUnleashed/0672329980_source/Chapter12/CS/MvcApplication1/MvcApplication1/Controllers/CompanyController.cs
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class CompanyController : Controller
    {
        [Authorize]
        public ActionResult Secrets()
        {
            return View();
        }

        [Authorize(Users="Jack,Jill")]
        public ActionResult SuperSecrets()
        {
            return View();
        }

        [Authorize(Roles = "Manager")]
        public ActionResult SuperSuperSecrets()
        {
            return View();
        }


    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter12/VB/MvcApplication1/MvcApplication1/Controllers/CompanyController.vb
Public Class CompanyController
    Inherits Controller

    <Authorize()> _
    Public Function Secrets() As ActionResult
        Return View()
    End Function

    <Authorize(Users:="Jack,Jill")> _
    Public Function SuperSecrets() As ActionResult
        Return View()
    End Function

    <Authorize(Roles:="Manager")> _
    Public Function SuperSuperSecrets() As ActionResult
        Return View()
    End Function

End Class
